demo.cc

```diff
@@ -576,16 +579,134 @@ class Cruncher final {
   void operator=(Cruncher&&) = delete;
 };
 
+class MixpanelUploader final {
+ public:
+  MixpanelUploader(const std::string& demo_id, const std::string& mixpanel_token)
+      : demo_id_(demo_id), mixpanel_token_(mixpanel_token) {}
+
+  inline bool Entry(std::unique_ptr<schema::Base>& entry, size_t index, size_t total) {
+    static_cast<void>(index);
+    static_cast<void>(total);
+
+    struct types {
+      typedef schema::Base base;
+      typedef std::tuple<schema::AnswerRecord> derived_list;
+      typedef bricks::rtti::RuntimeTupleDispatcher<base, derived_list> dispatcher;
+    };
+    types::dispatcher::DispatchCall(*entry, *this);
+
+    return true;
+  }
+
+  inline void Terminate() { std::cerr << '@' << demo_id_ << " MixpanelUploader is done.\n"; }
+
+  inline void operator()(schema::Base&) { /* ignore */
+  }
+
+  inline void operator()(schema::AnswerRecord& a) {
+    std::cerr << '@' << demo_id_ << " MixpanelUploader +A: " << a.uid << " `" << static_cast<int>(a.answer)
+              << "` Q" << static_cast<size_t>(a.qid) << '\n';
+    MixpanelQuestionAnsweredEvent ev(mixpanel_token_, a);
+    // WORKAROUND(sompylasar): `bricks::cerealize::JSON` cannot make more than one top-level key-value pair but we need this to build Mixpanel requests.
+    const std::string json = MultiKeyJSON(ev);
+    std::cerr << '@' << demo_id_ << " MixpanelUploader Event: " << json << std::endl;
+    const std::string base64_json = Base64Encode(json);
+    // WORKAROUND(sompylasar): Not using `https://`, could not send HTTPS request.
+    const std::string mixpanel_request = "http://api.mixpanel.com/track?data=" + base64_json;
+    std::cerr << '@' << demo_id_ << " MixpanelUploader Request: " << mixpanel_request << std::endl;
+    if (mixpanel_token_.empty()) {
+      std::cerr << '@' << demo_id_ << " MixpanelUploader Empty token, not sending." << std::endl;
+      return;
+    }
+    auto response = HTTP(GET(mixpanel_request));
+    std::cerr << '@' << demo_id_ << " MixpanelUploader Response: HTTP " << static_cast<int>(response.code) << " \"" << response.body << "\"" << std::endl;
+  }
+
+  template <typename T>
+  inline std::string MultiKeyJSON(T& object) {
+    std::ostringstream os;
+    {
+      // This scope is for `cereal` to flush the archive on scope exit.
+      auto ar =
+          bricks::cerealize::CerealStreamType<bricks::cerealize::CerealFormat::JSON>::CreateOutputArchive(os);
+      // The following allows to make more than one top-level key-value pair.
+      object.serialize(ar);
+    }
+    return os.str();
+  }
+
+  inline std::string Base64Encode(const std::string& str) {
+    // Note: Using `cereal`'s third-party library for base64-encoding.
+    return base64::encode(reinterpret_cast<const unsigned char*>(str.c_str()), str.length());
+  }
+
+  struct MixpanelQuestionAnsweredEvent {
+    struct Properties {
+      // Reserved properties.
+      std::string token;
+      std::string distinct_id;
+      time_t time;
```

Comment:
Let's use a C++ type, like `uint64_t`, `uint32_t`, `double`, etc. `time_t` is too unclear and possibly platform-dependent.